### PR TITLE
parse the puzzle reveal once, instead of 3 times

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: chia/wallet/wallet_state_manager.py [hunk 1]
```diff
@@ -634,24 +634,25 @@ async def determine_coin_type(
...
         # First spend where 1 mojo coin -> Singleton launcher -> NFT -> NFT
         try:
-            uncurried_nft = UncurriedNFT.uncurry(Program.from_bytes(bytes(coin_spend.puzzle_reveal)))
+            uncurried_nft = UncurriedNFT.uncurry(puzzle)
+            return await self.handle_nft(coin_spend, uncurried_nft)
         except Exception:
             # This is not a NFT coin, skip NFT handling
             pass
-        else:
-            return await self.handle_nft(coin_spend, uncurried_nft)
```
Comment (arvidn) [inline chia/wallet/wallet_state_manager.py:651]: I moved this up into the `try`-block, as I think it simplifies the code (and saves one line)

Comment (mariano54) [reply chia/wallet/wallet_state_manager.py:651]: yep, else is confusing, it's not executed in exceptions, but it's not clear

**Review by mariano54 — APPROVED** [review]

**Review by jack60612 — APPROVED** [review]
less lines, but how much faster is it?

**Review by trepca — APPROVED** [review]

**Comment by arvidn** [comment]
> but how much faster is it?

I don't have a good measurement of the speed-up. It's just a thing that shows up in my profiles. Ideally we would have `uncurry()` that would work on `SerializedProgram` directly, so we wouldn't even have to parse it into a python structure

**Pull request merged**